test/GPv2TradeSimulator.test.ts

```diff
@@ -0,0 +1,233 @@
+import IERC20 from "@openzeppelin/contracts/build/contracts/IERC20.json";
+import IUniswapV2Router from "@uniswap/v2-periphery/build/IUniswapV2Router02.json";
+import { expect } from "chai";
+import { MockContract } from "ethereum-waffle";
+import { Contract } from "ethers";
+import { artifacts, ethers, waffle } from "hardhat";
+
+import { InteractionLike, InteractionStage, TradeSimulator } from "../src/ts/";
+
+function mockReturnsInteraction(
+  mock: MockContract,
+  method: string,
+  params: unknown[],
+  result: unknown[],
+): InteractionLike {
+  const doppelganger = new ethers.utils.Interface([
+    "function __waffle__mockReturns(bytes memory data, bytes memory value)",
+  ]);
+  return {
+    target: mock.address,
+    callData: doppelganger.encodeFunctionData("__waffle__mockReturns", [
+      mock.interface.encodeFunctionData(method, params),
+      mock.interface.encodeFunctionResult(method, result),
+    ]),
+  };
+}
+
+describe("GPv2TradeSimulator", () => {
+  const [deployer, owner, trader, receive] = waffle.provider.getWallets();
+  let settlement: Contract;
+  let tradeSimulator: TradeSimulator;
+  let tokens: [MockContract, MockContract];
+  let router: MockContract;
+
+  beforeEach(async () => {
+    const GPv2AllowListAuthentication = await ethers.getContractFactory(
+      "GPv2AllowListAuthentication",
+      deployer,
+    );
+    const authenticator = await GPv2AllowListAuthentication.deploy();
+    await authenticator.initializeManager(owner.address);
+
+    const IVault = await artifacts.readArtifact("IVault");
+    const vault = await waffle.deployMockContract(deployer, IVault.abi);
+
+    const GPv2Settlement = await ethers.getContractFactory(
+      "GPv2SettlementTestInterface",
+      deployer,
+    );
+    settlement = await GPv2Settlement.deploy(
+      authenticator.address,
+      vault.address,
+    );
```

Comment:
Why don't we use our standard deployment function that we use for end to end tests?

Reply:
I don't consider this an E2E test since all interactions and tests rely on mocking all contracts.